### ISSUE TITLE
Pod install error related to custom project configuration names

### ISSUE THREAD

**Issue by Brett-Best** [CocoaPods/CocoaPods#9465]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Changed the project configuration names from `Debug` -> `Debug - Production` and `Release` -> `Release - Production`, this causes the pod install issue. Only occurs in the beta version of Cocoapods.
Ran `pod install`.

## What did you expect to happen?

Install all pod dependencies correctly.

## What happened instead?

```
NoMethodError - undefined method `merged_pod_target_xcconfigs' for nil:NilClass
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:166:in `block in remove_pod_target_xcconfig_overrides_from_target'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:164:in `each'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:164:in `remove_pod_target_xcconfig_overrides_from_target'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:147:in `block in add_target'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:146:in `tap'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:146:in `add_target'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:49:in `block in install!'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/user_interface.rb:145:in `message'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:34:in `install!'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:114:in `block in install_pod_targets'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:111:in `map'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:111:in `install_pod_targets'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/single_pods_project_generator.rb:32:in `block in install_all_pod_targets'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/user_interface.rb:145:in `message'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/single_pods_project_generator.rb:31:in `install_all_pod_targets'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer/xcode/single_pods_project_generator.rb:19:in `generate!'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer.rb:308:in `block in create_and_save_projects'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/user_interface.rb:64:in `section'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer.rb:303:in `create_and_save_projects'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer.rb:295:in `generate_pods_project'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer.rb:174:in `integrate'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/installer.rb:162:in `install!'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/command/install.rb:52:in `run'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/lib/cocoapods/command.rb:52:in `run'
/Users/brettbest/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/cocoapods-1.9.0.beta.2/bin/pod:55:in `<top (required)>'
/Users/brettbest/.rbenv/versions/2.6.3/bin/pod:23:in `load'
/Users/brettbest/.rbenv/versions/2.6.3/bin/pod:23:in `<main>'
```

### Stack

```
   CocoaPods : 1.9.0.beta.2
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin19]
    RubyGems : 3.1.2
        Host : Mac OS X 10.15.3 (19D49f)
       Xcode : 11.3 (11C29)
         Git : git version 2.24.1
Ruby lib dir : /Users/brettbest/.rbenv/versions/2.6.3/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/brettbest/.rbenv/versions/2.6.3/bin/pod
```

### Plugins

```
cocoapods-acknowledgements : 1.1.3
cocoapods-deintegrate      : 1.0.4
cocoapods-plugins          : 1.0.0
cocoapods-search           : 1.0.0
cocoapods-stats            : 1.1.0
cocoapods-trunk            : 1.4.1
cocoapods-try              : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'CP-Bug' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for CP-Bug
  pod 'Alamofire'
end
```

## Project that demonstrates the issue

https://github.com/Brett-Best/CP-Bug

**Comment by Brett-Best** [comment]
Suspect this is related to https://github.com/CocoaPods/CocoaPods/issues/9461.

**Comment by dnkoutso** [comment]
Fix was merged, will ship with 1.9.0.beta.3

**Comment by sweetleon** [comment]
@dnkoutso  Any word on when `1.9.0.beta.3` will be available? This is blocking my project, and presumably others.
